timeslotnotifications.php

```diff
@@ -0,0 +1,91 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Timeslot notifications management page.
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>, Jack Kepper <Jack@Kepper.net>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once(__DIR__.'/../../config.php');
+
+$id = required_param('id', PARAM_INT); // Observation instance ID.
+$action = optional_param('action', null, PARAM_TEXT);
+$notifyid = optional_param('notifyid', null, PARAM_INT);
+
+list($observation, $course, $cm) = \mod_observation\observation_manager::get_observation_course_cm_from_obid($id);
+$pageurl = new moodle_url('/mod/observation/timeslotnotifications.php', ['id' => $id]);
+
+// Check permissions.
+require_login($course, true, $cm);
+
+// Check for actions in URL params.
+if ($action !== null) {
+
+    if ($notifyid === null) {
+        throw new \moodle_exception('missingparam', 'error', null, $a = 'notifyid');
+    }
+
+    switch($action) {
+        case 'delete':
+            require_sesskey();
+            \mod_observation\timeslot_manager::delete_notification($observation->id, $USER->id, $notifyid);
+            redirect($pageurl);
+        break;
+    }
+}
+
+// Get current timeslot user registered to.
+$timeslot = \mod_observation\timeslot_manager::get_registered_timeslot($observation->id, $USER->id);
+
+// Load forms.
+$prefill = [
+    'id' => $observation->id,
+    'slotid' => $timeslot->id
+];
+$notificationeditor = new \mod_observation\form\notificationeditor(null, $prefill);
+
+if ($fromform = $notificationeditor->get_data()) {
+    require_sesskey();
```

Comment:
I think get_data() has an implicit sesskey check inside it, so the require sesskey is likely not required here. Formslib should make this call redundant